src/s390x/zipl.rs

```diff
@@ -33,6 +34,69 @@ pub fn chreipl<P: AsRef<Path>>(dev: P) -> Result<()> {
     Ok(())
 }
 
+/// Sets zVM secure loaddev to `dev`.
+pub fn set_loaddev<P: AsRef<Path>>(dev: P) -> Result<()> {
+    eprintln!("Setting LOADDEV");
+    let output = runcmd_output!(
+        "lszdev",
+        "-n",
+        "--columns",
+        "TYPE,ID",
+        "--by-node",
+        dev.as_ref()
+    )?;
+    let (devtype, id) = output
+        .trim()
+        .split_once(' ')
+        .with_context(|| format!("parsing {output}"))?;
+
+    let mut cmd = Command::new("vmcp");
+    cmd.arg("set").arg("loaddev");
+    match devtype {
+        "dasd-eckd" => {
+            lazy_static! {
+                static ref REGEX: Regex =
+                    Regex::new(r#"[[:digit:]].[[:digit:]].([[:xdigit:]]+)"#).unwrap();
+            }
+            for cap in REGEX.captures_iter(id) {
+                cmd.arg("eckd").arg("dev").arg(&cap[1]);
+            }
+        }
+        "zfcp-lun" => {
+            lazy_static! {
+                static ref REGEX: Regex = Regex::new(
+                    r#"[[:digit:]].[[:digit:]].([[:xdigit:]]+):0x([[:xdigit:]]+):0x([[:xdigit:]]+)"#
+                )
+                .unwrap();
+            }
+            // CP tool wants portname/lun to be splitted at 8th character:
+            // $ vmcp set loaddev dev 8007 portname 500507630400d1e3 lun 4001404c00000000 secure
+            // HCPZPM002E Invalid operand - 500507630400D1E3
+
+            // $ vmcp set loaddev dev 8007 portname 50050763 0400d1e3 lun 4001404c 00000000 secure
+            // $ vmcp q loaddev
+            // SCSI     DEVICE   8007 SECURE
+            // PORTNAME 50050763 0400D1E3    LUN  4001404C 00000000    BOOTPROG 0
+            // BR_LBA   00000000 00000000
+
+            for cap in REGEX.captures_iter(id) {
+                cmd.arg("dev")
+                    .arg(&cap[1])
+                    .arg("portname")
+                    .arg(&cap[2][0..8])
+                    .arg(&cap[2][8..])
+                    .arg("lun")
+                    .arg(&cap[3][0..8])
+                    .arg(&cap[3][8..]);
+            }
+        }
+        _ => bail!("unsupported device: {} id: {}", devtype, id),
```

Comment:
Ouf, that's unfortunate. Any interface in e.g. `/sys` we could use instead?

Reply:
Do you mean using `/sys` to get dev/lun/port? Yep, it's possible, but requires a lot of code, which would be similar to what `lszdev` has/does. I was thinking about it, but we already use `lsblk` , so decided to use existing tool to get exact ids of `install_target`

Reply:
OK.

Can you (1) factor out the parsing into separate functions, e.g. `parse_lszdev_dasd` and `parse_lszdev_zfcp`, and (2) in those functions, provide sample output in a comment of what we're trying to parse?